src/appleseed/renderer/modeling/environmentedf/oslenvironmentedf.cpp

```diff
@@ -63,10 +91,81 @@ namespace renderer
 
 namespace
 {
+
+    typedef ImageImportanceSampler<Color3f, float> ImageImportanceSamplerType;
+
     //
-    // OSL environment EDF.
+    // OSL environment sampler.
     //
-    // todo: importance sampling
+
+    class OSLEnvironmentSampler
+    {
+      public:
+
+        OSLEnvironmentSampler(
+            OSLShadingSystem* osl_shading_system,
+            ShaderGroup* shader_group,
+            Arena& arena,
+            const size_t width,
+            const size_t height)
+          : m_osl_shading_system(osl_shading_system)
+          , m_shader_group(shader_group)
+          , m_arena(arena)
+          , m_width(width)
+          , m_height(height)
+          , m_rcp_width(1.0f / width)
+          , m_rcp_height(1.0f / height)
+          , m_osl_thread_info(osl_shading_system->create_thread_info())
```

Comment:
I think it would be better to make OSLShaderGroupExec::execute_background public instead or repeating code.

Reply:
Making OSLShaderGroupExec::execute_background() public seems as a good choice for this problem. But all other OSLShaderGroupExec class methods are private so it seemed to me that I would "break" the consistency by making OSLShaderGroupExec::execute_background() public. 
So if you agree, I will make OSLShaderGroupExec::execute_background() public.

Reply:
Yes, make it public. It is better than reimplenting the code.

Reply:
Do I understand correctly that this has yet to be done?

Reply:
Yes, this still has to be done. I am running some tests now it should be up today.

Reply:
Done. Thanks for the comment.